### PR TITLE
Encryption WIP

### PR DESCRIPTION
**Pull request by sebastianharko**
Soliciting early feedback so feel free to comment here.

### CODE REVIEW

**Comment by bekiroguz** [comment]
Looks very clean, thanks. Maybe we can also add a turn on/off switch for encryption.

File: akka/src/main/scala/io/kagera/akka/actor/AkkaObjectSerializer.scala [hunk 1]
```diff
@@ -14,7 +14,7 @@ class AkkaObjectSerializer(system: ActorSystem) extends ObjectSerializer {
     // for now we re-use akka Serialization extension for pluggable serializers
     val serializer = serialization.findSerializerFor(obj)
 
-    val bytes = serializer.toBinary(obj)
+    val bytes = Encryption.AES.encrypt(serializer.toBinary(obj), system.settings.config.getString("encryption.secret"))
```
Comment (merlijn) [inline akka/src/main/scala/io/kagera/akka/actor/AkkaObjectSerializer.scala]: Can we make the encryption type "AES" or "DES" configurable?

File: akka/src/main/resources/reference.conf [hunk 2]
```diff
@@ -10,4 +10,7 @@ akka {
       "io.kagera.persistence.messages.Initialized" = kagera-scalapb
     }
   }
+  encryption {
+    secret = "0123456789012345"
```
Comment (merlijn) [inline akka/src/main/resources/reference.conf]: Should maybe move under kagera { .. } ?

**Comment by sebastianharko** [comment]
@bekiroguz I added an on/off switch for encryption in the sense that you can initialize 
the serializer using 

`new Serialization(new AkkaObjectSerializer(context.system, encryption = NoEncryption))`

or using 

`new Serialization(new AkkaObjectSerializer(context.system, encryption = AESJavaCryptoEncryption))`

**Comment by sebastianharko** [comment]
@merlijn I'm reading the encryption secret using ConfigFactory.load().getString("kagera.encryption.secret") in Encryption.scala. ConfigFactory.load() is supposed to have caching so there's little or no performance penalty. Is this ok with you or you'd rather I get a reference to configuration/settings from some place else ?

**Pull request merged**